### PR TITLE
refactor(st1page): improve some code and cherry-pick

### PR DESCRIPTION
**Pull request by ice1000**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Modified some code inspired from #6145

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#6145

### CODE REVIEW

File: src/frontend/src/optimizer/property/distribution.rs [hunk 1]
```diff
@@ -328,20 +347,27 @@ impl RequiredDist {
...
+    }
+
+    fn to_dist(&self, plan: &impl GenericPlanRef) -> Distribution {
         let dist = match self {
+            // all the distribution satisfy the Any, and the function can be only called by
+            // `enforce_if_not_satisfies`
             RequiredDist::Any => unreachable!(),
             // TODO: add round robin distributed type
-            RequiredDist::AnyShard => todo!(),
+            RequiredDist::AnyShard => Distribution::HashShard(plan.logical_pk().to_vec()),
```
Comment (st1page) [inline src/frontend/src/optimizer/property/distribution.rs:364]: prefer to leave a todo!(), for actually the brach is not achieved now. And if the logical_pk() is empty, maybe it is wrong.

**Review by st1page — APPROVED** [review]
rest LGTM

File: src/frontend/src/optimizer/property/distribution.rs (hunk 1 above)
Comment (ice1000) [reply src/frontend/src/optimizer/property/distribution.rs:364]: ura!

Comment (ice1000) [reply src/frontend/src/optimizer/property/distribution.rs]: fixed

**Pull request merged**